frontend/models/product-list/server.ts

```diff
@@ -401,3 +424,27 @@ function createPublicAlgoliaKey(appId: string, apiKey: string): string {
    });
    return publicKey;
 }
+
+async function findDevicesWithProducts(devices: string[]) {
+   return timeAsync('algolia.findDescendantDevicesWithProducts', async () => {
+      const client = algoliasearch(
+         ALGOLIA_APP_ID,
+         ALGOLIA_API_KEY,
+         CLIENT_OPTIONS
+      );
+      const index = client.initIndex(ALGOLIA_PRODUCT_INDEX_NAME);
+      const deviceDisjunctiveFilters = devices
+         .map((device) => `device:"${escapeFilterValue(device)}"`)
+         .join(' OR ');
+      const deviceFilterSuffix = deviceDisjunctiveFilters
+         ? ` AND ${deviceDisjunctiveFilters}`
+         : '';
+      const { facets } = await index.search('', {
+         facets: ['device'],
+         filters: `public = 1${deviceFilterSuffix}`,
+         maxValuesPerFacet: 1000,
+         hitsPerPage: 0,
```

Comment:
Ask for 0 hits, sneaky!

Reply:
Yep! That way the only data we get back is the list of devices, instead of all the json included in the product hits